### DISCUSSION TITLE
How to link to part of the same document in Markdown?

### DISCUSSION THREAD

**Discussion by joe-berg-colossus** [ueberdosis/tiptap#879] [General Questions]
Looking to link to a specific section in the same document using standard markdown syntax shown below:

```
[Custom foo description](#foo)

# Foo
```

Is this possible?

**Comment by andreasvirkus** [comment]
I think you'd have to build a custom heading extension, namely expanding the schema of the [current Heading component](https://github.com/ueberdosis/tiptap/blob/main/packages/tiptap-extensions/src/nodes/Heading.js) so that it renders the Node with an ID attribute, based on its contents.

Here's an example with generating unique ID hashes - https://github.com/ueberdosis/tiptap/issues/621

Maybe @hanspagel can help us out with an example 🙏  - I'm not sure how to access the node's textual value inside `toDom` 🤔

**Reply by andreasvirkus** [reply]
Linking another relevant issue - https://github.com/ueberdosis/tiptap/issues/662
